repository_pool.go

```diff
@@ -25,34 +25,53 @@ func NewRepository(id string, repo *git.Repository) Repository {
 	}
 }
 
-// RepositoryPool holds a pool of initialized git repositories and
+// NewRepositoryFromPath creates and initializes a new Repository structure
+// and initializes a go-git repository
+func NewRepositoryFromPath(id, path string) (Repository, error) {
+	repo, err := git.PlainOpen(path)
+	if err != nil {
+		return Repository{}, err
+	}
+
+	return NewRepository(id, repo), nil
+}
+
+// RepositoryPool holds a pool git repository paths and
 // functionality to open and iterate them.
 type RepositoryPool struct {
-	repositories []Repository
+	repositories map[string]string
```

Comment:
what is the key and the value to this? I saw in the tests the key is the repo path, but what's the value?

Reply:
The key is the ID. The idea is not to use the full path as an ID. For example if we have the path `repos/00/001122334455` the ID could be `001122334455`. This is the same ID that appears as `repository_id` column.

Reply:
What if we have `/repos/foo` and `/repos/bar/foo`? Isn't that a bit misleading? Not everyone will have the repositories bucketed, so name clashes may happen.

Reply:
We can either change `AddGit` to add another number of directories or set the `ID` manually:

```
pool.Add("some_fancy_identifier", "/storage/bricks/83/repos/846283")
```

Truncating the name to create the ID is just quick fix for now to have a `repository_id` that is not the full path. In the future that identifier should be unique and really identify the repo. For example with a UUID. If we use the full path and the repository is moved the the ID will change and this may lead to problems.

I can set the full path for now if you think this will be less error prone.

Reply:
Now ID is the full path.